.up/config/datadog-xm/Dockerfile

```diff
@@ -0,0 +1,7 @@
+FROM gcr.io/datadoghq/agent:latest
```

Comment:
can we switch Dockerfiles and files to cluster/images/<image-name>/* 
like: https://github.com/crossplane-contrib/x-metrics/tree/main/cluster/images/x-metrics

and please add the following section in makefile:
https://github.com/crossplane-contrib/x-metrics/blob/main/Makefile#L47-L54

then the docker build process is used from upbound build module